django-stubs/core/exceptions.pyi

```diff
@@ -25,27 +30,37 @@ class MiddlewareNotUsed(Exception): ...
 class ImproperlyConfigured(Exception): ...
 class FieldError(Exception): ...
 
-NON_FIELD_ERRORS: Literal["__all__"]
+NON_FIELD_ERRORS: Literal["__all__"] = ...
+
+_MsgTypeBase = Union[str, ValidationError]
+# Yeah, it's really ugly, but __init__ checks with isinstance()
+_MsgType = Union[
+    _MsgTypeBase,
+    Dict[str, _MsgTypeBase],
+    List[_MsgTypeBase],
+    Dict[str, str],
+    Dict[str, ValidationError],
+    List[str],
+    List[ValidationError],
```

Comment:
This is not correct. `ValidationError` accepts arbitrarily nested data structures as the `message` argument.

For example this correct code from `django-rest-framework` produces a false positive error: https://github.com/encode/django-rest-framework/blob/3.13.1/tests/test_fields.py#L2428-L2433

Mypy does not support recursive type aliases, so it's impossible to describe this type accurately AFAIK.

For now I think it's preferable to revert than to have false positives. I have opened PR #943 for that. If you have any thoughts, please comment there.